core/src/main/scala/org/broadinstitute/dsde/rawls/entities/base/InputExpressionReassembler.scala

```diff
@@ -169,17 +175,61 @@ object InputExpressionReassembler {
    */
   private def reconstructInputExprForEachEntity(mapOfEntityToEvaluatedExprMap: Map[EntityName, Try[Map[LookupExpression, JsValue]]],
                                                 parsedTree: ParseTree,
-                                                rootEntityNames: Option[Seq[EntityName]]): Map[EntityName, Try[JsValue]] = {
+                                                rootEntityNames: Option[Seq[EntityName]],
+                                                inputOption: Option[MethodInput]): Map[EntityName, Try[JsValue]] = {
+
+    def updateInputExprValueMapForObjectInput(inputExprWithEvaluatedRef: JsValue, objectFieldsFromInput: mutable.Seq[ValueTypeObjectFieldTypes]): JsObject = {
+      val updatedObject: Map[String, JsValue] = inputExprWithEvaluatedRef.asJsObject.fields.map { case (inputName, evaluatedValue) =>
```

Comment:
I'm finding this difficult to read/understand. Why does there need to be a case statement (don't we do this for all keys/values)?

Reply:
Yes. We are iterating for all (key, value) pairs inside the map. Since the data type of fields inside a JsObject is Map[String, JsValue], we can use a case statement to identify the key and value separately. This can also be written as below, but using case statement and having explicit variable names makes it more readable. In my opinion, using `obj._1` and `obj._2` makes it harder to understand/read what is going on. We can always extract `obj._1` and `obj._2` into variables but case statement implicitly does that.
```
inputExprWithEvaluatedRef.asJsObject.fields.map { obj =>
// in this case obj._1 will be key of type String and obj._2 will be the value of type JsValue

val inputObjectField: Option[ValueTypeObjectFieldTypes] = objectFieldsFromInput.find(x => x.getFieldName == obj._1)
...
}
```